### PR TITLE
syslogng: Late cleanup

### PR DESCRIPTION
**Pull request by vifino**
## Description of changes
Late review of #257696 made me dig deeper into how the build process uses pip.
This addresses the two points raised and also bumps the version in the module so it doesn't emit a warning.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vifino** [comment]
@SuperSandro2000 I've experienced build issues on aarch64-linux without this patch as pip is trying to be a smartass even when it has no network connectivity, please merge this.

File: pkgs/tools/system/syslog-ng/default.nix [hunk 1]
```diff
@@ -100,7 +99,7 @@ stdenv.mkDerivation rec {
     "--enable-dynamic-linking"
     "--enable-systemd"
     "--enable-smtp"
-    "--with-python-packages=system"
```
Comment (symphorien) [inline pkgs/tools/system/syslog-ng/default.nix:103]: I'm curious about the difference this makes: it seems to have the same consequences in https://github.com/syslog-ng/syslog-ng/blob/449f16278a61ff83e069190b4850002d7c7c68d7/configure.ac#L1740-L1749

Comment (vifino) [reply pkgs/tools/system/syslog-ng/default.nix:103]: At some point, I think `modules/python-modules/Makefile.am` tries to run setup.py and pip, unless `[ "$(with_python_packages)" != "none" ]`.

Comment (symphorien) [reply pkgs/tools/system/syslog-ng/default.nix:103]: did you check that it installs as much, as opposed to disabling building/installing python modules ?

Comment (vifino) [reply pkgs/tools/system/syslog-ng/default.nix:103]: It does install the python modules in `lib/syslog-ng/python` as far as I can tell.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**